osu.Game/OsuGame.cs

```diff
@@ -432,11 +438,47 @@ public bool OnPressed(GlobalAction action)
                 case GlobalAction.ToggleDirect:
                     direct.ToggleVisibility();
                     return true;
+                case GlobalAction.TakeScreenshot:
+                    if (Window.ScreenshotTakenAction == null)
+                        Window.ScreenshotTakenAction = (screenshotBitmap) =>
+                        {
+                            var fileName = getScreenshotFileName(screenshotFormat);
+
+                            switch (screenshotFormat.Value)
+                            {
+                                case ScreenshotFormat.Bmp:
+                                    screenshotBitmap.Save(fileName, ImageFormat.Bmp);
+                                    break;
+                                case ScreenshotFormat.Png:
+                                    screenshotBitmap.Save(fileName, ImageFormat.Png);
+                                    break;
+                                case ScreenshotFormat.Jpg:
+                                    screenshotBitmap.Save(fileName, ImageFormat.Jpeg);
+                                    break;
+                                default:
+                                    throw new ArgumentOutOfRangeException(nameof(screenshotFormat));
+                            }
+                        };
+
+                    RequestScreenshot();
+                    return true;
             }
 
             return false;
         }
 
+        private string getScreenshotFileName(ScreenshotFormat screenshotFormat)
+        {
+            // TODO Change screenshots location

```

Comment:
todo

Reply:
I would like somebody from the osu! team to tell me where to store screenshots. It is not obvious to me because at the moment I have osu! lazer as two separate folders. The first folder is "bin" which contains build artifacts like *.exe and *.dll files. The second folder is %appdata%/osu! which contains game/framework configs, beatmaps and so on.

Reply:
it should be stored alongside all other use data.